Support/lib/ruby_utils.rb

```diff
@@ -0,0 +1,55 @@
+module RubyUtils
+  module_function
+
+  # Try to find the Ruby executable `name`. Return an array containing the
+  # command(s) needed to run the executable, or `nil` if the executable could
+  # not be found.
+  #
+  # Typical usage looks like this:
+  #
+  #     Dir.chdir(ENV['TM_PROJECT_DIRECTORY']) if ENV['TM_PROJECT_DIRECTORY']
+  #     executable = RubyUtils.find_executable('rspec')
+  #     if executable
+  #       system(*executable, '--some', '--additional', 'args')
+  #     else
+  #       # Executable not found, so use fallback / display alert / …
+  #     end
+  #
+  # Supports the following cases:
+  #
+  # 1. If an appropriate `TM_*` environment variable is present and points to an
```

Comment:
> Also, to use this for both projects with a Gemfile and without one, I would need to constantly activate and disable the environment variable

We can add a “project scope” for projects with a `Gemfile` so that we can target these (or target projects without), here’s the list of current project attributes: https://github.com/textmate/textmate/blob/master/Frameworks/DocumentWindow/src/DocumentController.mm#L1429

It is true that users are not unlikely to set (globally) environment variables and forget about them, and then later something breaks because of this, but I think that goes for pretty much all environment variables.

Not having it trump everything means that if a user reports that for some project, our logic is wrong, we cannot provide them with any workaround, they’ll have to wait for someone to fix the code and push a new release, that’s IMO where the environment variable comes in: to allow users to fix things locally.

Using it as a fallback, IMHO that is the job of the `PATH` variable. I.e. ensure that whatever fallback executable you want to use when there is no `TM_«executable»`, no `Gemfile`, no etc., can be found via `PATH`.